### DISCUSSION TITLE
Strange Stress Concentrations on Periodic RVE

### DISCUSSION THREAD

**Discussion by Zoophish** [idaholab/moose#23830] [Q&A Modules: Solid mechanics] [answered]
Hi,

I'm running a small strain model of an RVE using the homogenisation system in the new tensor mechanics system. I've noticed I get these unphyiscal looking stress concentrations on the boundaries. I'm following the test case given in [this](https://www.sciencedirect.com/science/article/pii/S0168874X20301165) paper and it is clear that their model does not give these stress concentrations. I am using TET4 elements instead of HEX8 as I cannot mesh hex elements.

The artifact (this example is with the xx homogenisation stress constraint):
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/3082590/227283497-f51c1779-f804-4f3b-ba0e-2728693f7825.png">

Rescaling the colour bar reveals the stresses (around the corners) which roughly match the paper I am following:
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/3082590/227286120-2144206a-78df-420c-a695-b399d234014c.png">

For clarity, it should look something like this:
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/3082590/227285776-c6edb5eb-2e2f-4a36-8c86-c4282a7a04a1.png">

My input file and mesh file is here:
[input_file_mesh.zip](https://github.com/idaholab/moose/files/11053741/input_file_mesh.zip)

If this looks familiar, any help or pointers on what could be causing this would be greatly appreciated.

**Comment by GiudGiud** [comment, marked as answer]
The paper's mesh looks finer and more regular.
Could you please refine your mesh to see if the issue disappears?
There's a uniform refinement option in MOOSE if you dont want to go back to your meshing software

**Reply by Zoophish** [reply]
I reduced the element size in gmsh and it seems have fixed the artefact on my validation case.
<img width="50%" alt="image" src="https://user-images.githubusercontent.com/3082590/227505527-ed320d9b-9919-4481-83e0-01278409d6f8.png">

Thanks for your help!

**Comment by hugary1995** [comment]
TET4 is a bad choice for solid mechanics...
